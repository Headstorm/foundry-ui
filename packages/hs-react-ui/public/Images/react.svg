<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
