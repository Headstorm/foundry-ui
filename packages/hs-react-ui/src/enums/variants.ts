enum variants {
  fill = 'fill',
  outline = 'outline',
  text = 'text',
}

export default variants;
