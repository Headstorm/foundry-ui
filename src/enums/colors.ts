enum colors {
  background = '#fff',

  primary = '#DA4200',
  primaryDark = '#BF3A00',
  secondary = '#7349AC',
  secondaryDark = '#512888',
  tertiary = '#64ACC5',
  tertiaryDark = '#338BAA',

  black = '#000',
  grayDark = '#252D34',
  grayMedium = '#51575D',
  grayLight = '#9FA2A5',
  grayXlight = '#D5D5D5',

  success = '#698F00',
  destructive = '#d00',
  shadow = '#14161F',

  grayDark25 = 'rgba(211, 214, 215, 0.25)',
  grayDark50 = 'rgba(211, 214, 215, 0.5)',
  grayDark75 = 'rgba(211, 214, 215, 0.75)',

  transparent = 'transparent',
}

export default colors;
