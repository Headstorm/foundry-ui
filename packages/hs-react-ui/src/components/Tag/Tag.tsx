import React, { ReactNode } from 'react';
import UnstyledIcon from '@mdi/react';
import { mdiLoading } from '@mdi/js';
import styled, { StyledComponentBase } from 'styled-components';
import timings from '../../enums/timings';
import { useTheme } from '../../context';
import variants from '../../enums/variants';
import Progress from '../Progress/Progress';
import { Div } from '../../htmlElements';
import { getFontColorFromVariant, getBackgroundColorFromVariant } from '../../utils/color';
import { SubcomponentPropsType } from '../commonTypes';
import { getShadowStyle } from '../../utils/styles';

export type TagContainerProps = {
  elevation: number;
  color: string;
  variant: variants;
  type: string;
  disabled: boolean;
};

export type IconContainerProps = {
  hasContent: boolean;
  position: 'right' | 'left';
};

export type TagProps = {
  StyledContainer?: string & StyledComponentBase<any, {}>;
  containerProps?: SubcomponentPropsType;
  iconPrefix?: string | JSX.Element;
  iconSuffix?: string | JSX.Element;
  isLoading?: boolean;
  isProcessing?: boolean;
  children?: ReactNode;
  elevation?: number;
  variant?: variants;
  color?: string;
  StyledLoadingBar?: string & StyledComponentBase<any, {}>;
  loadingBarProps?: SubcomponentPropsType;
  StyledIconContainer?: string & StyledComponentBase<any, {}>;
  iconPrefixContainerProps?: SubcomponentPropsType;
  iconSuffixContainerProps?: SubcomponentPropsType;
  id?: string;
};

export const Container: string & StyledComponentBase<any, {}, TagContainerProps> = styled(Div)`
  ${({ elevation = 0, color, variant }: TagContainerProps) => {
    const { colors } = useTheme();
    const backgroundColor = getBackgroundColorFromVariant(variant, color, colors.transparent);
    const fontColor = getFontColorFromVariant(variant, color, colors.background, colors.grayDark);

    return `
      display: inline-flex;
      font-size: 1em;
      padding: .75em 1em;
      border-radius: 0.25em;
      transition:
        background-color ${timings.fast},
        color ${timings.slow},
        outline ${timings.slow},
        filter ${timings.slow},
        box-shadow ${timings.slow};
      ${getShadowStyle(elevation, colors.shadow)}
      outline: 0 none;
      border: ${variant === variants.outline ? `1px solid ${color || colors.grayDark}` : '0 none;'};
      cursor: pointer;
      background-color: ${backgroundColor};
      color: ${fontColor};
      align-items: center;
    `;
  }}
`;

const StyledProgress = styled(Progress)`
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
`;

const IconContainer = styled(Div)`
  ${({ position, hasContent }: IconContainerProps) => {
    return `
    height: 1rem;
    ${hasContent ? `margin-${position === 'right' ? 'left' : 'right'}: 1em;` : ''}
  `;
  }}
`;

const Tag = ({
  StyledContainer = Container,
  containerProps = {},
  StyledIconContainer = IconContainer,
  iconPrefixContainerProps = {},
  iconSuffixContainerProps = {},
  StyledLoadingBar = StyledProgress,
  loadingBarProps = {},
  iconPrefix,
  iconSuffix,
  isLoading,
  isProcessing,
  children,
  elevation = 0,
  variant = variants.fill,
  color,
  id,
}: TagProps): JSX.Element => {
  const hasContent = Boolean(children);
  const { colors } = useTheme();
  const containerColor = color || colors.grayLight;
  // get everything we expose + anything consumer wants to send to container
  const mergedContainerProps = {
    id,
    elevation,
    color: containerColor,
    variant,
    ...containerProps,
  };

  return isLoading ? (
    <StyledContainer {...mergedContainerProps}>
      <StyledLoadingBar {...loadingBarProps} />
    </StyledContainer>
  ) : (
    <StyledContainer {...mergedContainerProps}>
      {!isProcessing &&
        iconPrefix &&
        (typeof iconPrefix === 'string' && iconPrefix !== '' ? (
          <StyledIconContainer
            hasContent={hasContent}
            position="left"
            {...iconPrefixContainerProps}
          >
            <UnstyledIcon path={iconPrefix} size="1rem" />
          </StyledIconContainer>
        ) : (
          <StyledIconContainer>{iconPrefix}</StyledIconContainer>
        ))}
      {isProcessing && (
        <StyledIconContainer hasContent={hasContent} position="left" {...iconPrefixContainerProps}>
          <UnstyledIcon path={mdiLoading} size="1rem" spin={1} />
        </StyledIconContainer>
      )}
      {children}

      {iconSuffix &&
        (typeof iconSuffix === 'string' ? (
          <StyledIconContainer
            hasContent={hasContent}
            position="right"
            {...iconSuffixContainerProps}
          >
            <UnstyledIcon path={iconSuffix} size="1rem" />
          </StyledIconContainer>
        ) : (
          <StyledIconContainer
            hasContent={hasContent}
            position="right"
            {...iconSuffixContainerProps}
          >
            {iconSuffix}
          </StyledIconContainer>
        ))}
    </StyledContainer>
  );
};

Tag.Container = Container;
Tag.LoadingBar = StyledProgress;
Tag.IconContainer = IconContainer;
export default Tag;
