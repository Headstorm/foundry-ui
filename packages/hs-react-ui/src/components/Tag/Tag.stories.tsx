import React from 'react';
import { select, text, boolean, number, color } from '@storybook/addon-knobs';
import { mdiMessage, mdiSend } from '@mdi/js';
import { storiesOf } from '@storybook/react';

import Tag from './Tag';
import colors from '../../enums/colors';
import variants from '../../enums/variants';

const options = {
  none: '',
  mdiMessage,
  mdiSend,
};

const design = {
  type: 'figma',
  url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=0%3A1',
};

const testId = `tag-${Math.floor(Math.random() * 100000)}`;
const containerProps = { 'data-test-id': testId };

storiesOf('Tag', module).add(
  'Basic Tag',
  () => {
    return (
      <Tag
        variant={select('variant', variants, variants.fill)}
        color={color('color', colors.primaryDark)}
        isLoading={boolean('isLoading', false)}
        elevation={number('elevation', 1)}
        isProcessing={boolean('isProcessing', false)}
        iconPrefix={select('iconPrefix', options, options.none)}
        iconSuffix={select('iconSuffix', options, options.none)}
        containerProps={containerProps}
      >
        {text('children', 'Default text')}
      </Tag>
    );
  },
  { design },
);
