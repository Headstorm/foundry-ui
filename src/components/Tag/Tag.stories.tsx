import React from 'react';
import { Story, Meta } from '@storybook/react';
import { mdiMessage, mdiSend } from '@mdi/js';

import Tag, { TagProps } from './Tag';
import colors from '../../enums/colors';
import variants from '../../enums/variants';

export const BasicTag: Story<TagProps> = args => <Tag {...args} />;
BasicTag.args = {
  variant: variants.outline,
  color: colors.primaryDark,
  isLoading: false,
  elevation: 0,
  isProcessing: false,
  iconPrefix: 'none',
  iconSuffix: 'none',
  containerProps: { 'data-test-id': `tag-${Math.floor(Math.random() * 100000)}` },
  children: 'Default text',
};

const iconOptions = {
  options: ['none', 'mdiMessage', 'mdiSend'],
  mapping: {
    none: '',
    mdiMessage,
    mdiSend,
  },
  control: {
    type: 'radio',
  },
};

export default {
  title: 'Tag',
  component: Tag,
  argTypes: {
    elevation: { control: { type: 'range', min: -5, max: 5, step: 1 } },
    iconPrefix: iconOptions,
    iconSuffix: iconOptions,
  },
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=0%3A1',
    },
  },
} as Meta;
