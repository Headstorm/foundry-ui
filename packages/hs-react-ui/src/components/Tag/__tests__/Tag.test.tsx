import React from 'react';
import { render, waitFor, configure } from '@testing-library/react';
import Icon from '@mdi/react';
import colors from '../../../enums/colors';
import variants from '../../../enums/variants';
import Tag from '../Tag';
import { mdiComment } from '@mdi/js';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);
configure({ testIdAttribute: 'data-test-id' });

describe('Tag', () => {
  it('shows loading text when provided', async () => {
    const testId = 'tag-1';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag isLoading={true} containerProps={containerProps}>
        Shows Loading
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Tag with non-default props variant and color', async () => {
    const testId = 'tag-2';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag
        color={colors.black}
        elevation={2}
        variant={variants.outline}
        containerProps={containerProps}
      >
        Variant and color
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Tag with text variant', async () => {
    const testId = 'tag-3';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag variant={variants.text} containerProps={containerProps}>
        text variant
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Tag with fill variant', async () => {
    const testId = 'tag-4';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag variant={variants.fill} containerProps={containerProps}>
        fill variant
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Tag with outline variant', async () => {
    const testId = 'tag-5';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag variant={variants.outline} containerProps={containerProps}>
        outline variant
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows LeftIconContainer when isProcessing', async () => {
    const testId = 'tag-6';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag isProcessing containerProps={containerProps}>
        processing
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows icons with type string', async () => {
    const testId = 'tag-7';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag iconSuffix={mdiComment} iconPrefix={mdiComment} containerProps={containerProps}>
        string icon props
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows icons', async () => {
    const testId = 'tag-8';
    const containerProps = { 'data-test-id': testId };
    const { container, getByTestId } = render(
      <Tag
        iconSuffix={<Icon path={mdiComment} />}
        iconPrefix={<Icon path={mdiComment} />}
        containerProps={containerProps}
      >
        shows icons
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('keeps the container the same when switching between isLoading and not isLoading', async () => {
    const testId = 'tag-9';
    const containerProps = { 'data-test-id': testId };
    const { getByTestId, rerender, asFragment } = render(
      <Tag isLoading={true} containerProps={containerProps}>
        Hello World!
      </Tag>,
    );

    await waitFor(() => getByTestId(testId));
    const loadingFragment = asFragment();

    rerender(<Tag containerProps={containerProps}>Hello World!</Tag>);
    await waitFor(() => getByTestId(testId));
    const loadedFragment = asFragment();

    // TODO: Use toMatchDiffSnapshot() between the fragments once we can figure out
    // how to make it use the jest-styled-components plugin
    expect(loadingFragment.firstChild).toMatchSnapshot();
    expect(loadedFragment.firstChild).toMatchSnapshot();
  });
  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      const component = <Tag></Tag>;
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
  describe('Ref tests', () => {
    it('containerRef.current should exist', async () => {
      const testId = 'tag-10';
      const containerProps = { 'data-test-id': testId };
      const ref = React.createRef<HTMLSpanElement>();
      const { getByTestId } = render(<Tag containerRef={ref} containerProps={containerProps} />);
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLSpanElement).toBeTruthy();
    });
    it('iconPrefixContainerRef.current should exist', async () => {
      const testId = 'tag-11';
      const containerProps = { 'data-test-id': testId };
      const ref = React.createRef<HTMLElement>();
      const { getByTestId } = render(
        <Tag
          iconPrefixContainerRef={ref}
          iconPrefix={<div>prefix</div>}
          containerProps={containerProps}
        />,
      );
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLElement).toBeTruthy();
    });
    it('iconSuffixContainerRef.current should exist', async () => {
      const testId = 'tag-12';
      const containerProps = { 'data-test-id': testId };
      const ref = React.createRef<HTMLElement>();
      const { getByTestId } = render(
        <Tag
          iconSuffixContainerRef={ref}
          iconSuffix={<div>suffix</div>}
          containerProps={containerProps}
        />,
      );
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLElement).toBeTruthy();
    });
    it('loadingBarRef.current should exist', async () => {
      const testId = 'tag-13';
      const containerProps = { 'data-test-id': testId };
      const ref = React.createRef<HTMLElement>();
      const { getByTestId } = render(
        <Tag loadingBarRef={ref} isLoading containerProps={containerProps} />,
      );
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLElement).toBeTruthy();
    });
  });
});
