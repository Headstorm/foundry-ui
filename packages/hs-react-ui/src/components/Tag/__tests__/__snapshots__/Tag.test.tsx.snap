// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag keeps the container the same when switching between isLoading and not isLoading 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background: linear-gradient( 45deg, rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75) ) repeat;
  background-size: 400% 100%;
  width: 6rem;
  height: 16px;
  border-radius: 9999px;
  -webkit-animation: ia-dwnO 8s linear infinite;
  animation: ia-dwnO 8s linear infinite;
  line-height: 0;
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
}

<div
  class="sc-AxjAm c0"
  color="#7c8186"
  data-test-id="tag-9"
  elevation="0"
>
  <div
    class="sc-AxjAm sc-fzpjYC c1"
  />
</div>
`;

exports[`Tag keeps the container the same when switching between isLoading and not isLoading 2`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="sc-AxjAm c0"
  color="#7c8186"
  data-test-id="tag-9"
  elevation="0"
>
  Hello World!
</div>
`;

exports[`Tag shows LeftIconContainer when isProcessing 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 1rem;
  margin-right: 1em;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-6"
    elevation="0"
  >
    <div
      class="sc-AxjAm c1"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <style>
          @keyframes spin { to { transform: rotate(360deg) } }
        </style>
        <g
          style="animation: spin linear 1s infinite; transform-origin: center;"
        >
          <path
            d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
            style="fill: currentColor;"
          />
          <rect
            fill="transparent"
            height="24"
            width="24"
          />
        </g>
      </svg>
    </div>
    processing
  </div>
</div>
`;

exports[`Tag shows Tag with fill variant 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-4"
    elevation="0"
  >
    fill variant
  </div>
</div>
`;

exports[`Tag shows Tag with non-default props variant and color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  -webkit-filter: drop-shadow(0rem 0.07082177831667665rem 0.5rem rgba(39,47,78,0.39738511585523156));
  filter: drop-shadow(0rem 0.07082177831667665rem 0.5rem rgba(39,47,78,0.39738511585523156));
  outline: 0 none;
  border: 1px solid #000;
  cursor: pointer;
  background-color: transparent;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#000"
    data-test-id="tag-2"
    elevation="2"
  >
    Variant and color
  </div>
</div>
`;

exports[`Tag shows Tag with outline variant 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 1px solid #7c8186;
  cursor: pointer;
  background-color: transparent;
  color: #7c8186;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-5"
    elevation="0"
  >
    outline variant
  </div>
</div>
`;

exports[`Tag shows Tag with text variant 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: transparent;
  color: #7c8186;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-3"
    elevation="0"
  >
    text variant
  </div>
</div>
`;

exports[`Tag shows icons 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 1rem;
}

.c2 {
  height: 1rem;
  margin-left: 1em;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-8"
    elevation="0"
  >
    <div
      class="sc-AxjAm c1"
    >
      <svg
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
          style="fill: currentColor;"
        />
      </svg>
    </div>
    shows icons
    <div
      class="sc-AxjAm c2"
    >
      <svg
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
          style="fill: currentColor;"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Tag shows icons with type string 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 1rem;
  margin-right: 1em;
}

.c2 {
  height: 1rem;
  margin-left: 1em;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-7"
    elevation="0"
  >
    <div
      class="sc-AxjAm c1"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
          style="fill: currentColor;"
        />
      </svg>
    </div>
    string icon props
    <div
      class="sc-AxjAm c2"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
          style="fill: currentColor;"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Tag shows loading text when provided 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #7c8186;
  color: #000;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background: linear-gradient( 45deg, rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75) ) repeat;
  background-size: 400% 100%;
  width: 6rem;
  height: 16px;
  border-radius: 9999px;
  -webkit-animation: ia-dwnO 8s linear infinite;
  animation: ia-dwnO 8s linear infinite;
  line-height: 0;
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
}

<div>
  <div
    class="sc-AxjAm c0"
    color="#7c8186"
    data-test-id="tag-1"
    elevation="0"
  >
    <div
      class="sc-AxjAm sc-fzpjYC c1"
    />
  </div>
</div>
`;
